### PR TITLE
[#44] - added naming conventions to readme

### PR DESCRIPTION
**Pull request by LyonFoster**
take a look at this and lmk if I should add anything. We can update as we go along

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # ARDVARC
 Team ARDVARC's github repository
 
+## Team Members (open a pr to put urs below)
+* Tim
+* Lyon
+
```
Comment (aidanwoodcu) [inline README.md:7]:
```suggestion
* Aidan

```

File: README.md [hunk 2]
```diff
@@ -11,8 +15,22 @@ Team ARDVARC's github repository
...
 7. Bother them until it goes through
 
-Please Put Your Name Below to indicate that you can do this!
-Tim
+## Vector and Frame Naming Convention
+#### Vectors
+* `vec_a2b_in_frame`
+* `vec` - dentotes it's a vector
+* `a2b` - denotes the vectors originates at `a` and terminates at `b`
+* `in_frame` - denotes the frame that the vector is expressed in
```
Comment (aidanwoodcu) [inline README.md:23]:
```suggestion
    * `vec` - dentotes it's a vector
    * `a2b` - denotes the vectors originates at `a` and terminates at `b`
    * `in_frame` - denotes the frame that the vector is expressed in
```

File: README.md [hunk 3]
```diff
@@ -11,8 +15,22 @@ Team ARDVARC's github repository
...
+
+#### Rotations
+* `object_a2b`
+* `object` - denotes the type of rotation parameter:
+    * `q` - for unit quaternion
+    * `mrp` - for modified rodrigues parameter
+    * `dcm` - for direction cosine matrix
+* `a2b` - denotes that the rotation parameter transforms a vector from being expressed in frame `a` to being expressed in frame `b`
```
Comment (aidanwoodcu) [inline README.md:31]:
```suggestion
    * `object` - denotes the type of rotation parameter:
        * `q` - for unit quaternion
        * `mrp` - for modified rodrigues parameter
        * `dcm` - for direction cosine matrix
    * `a2b` - denotes that the rotation parameter transforms a vector from being expressed in frame `a` to being expressed in frame `b`
```

File: README.md (hunk 2 above)
Comment (aidanwoodcu) [inline README.md:23]: Should we list what frames we're using? Do we know yet?

Comment (LyonFoster) [reply README.md:23]: we could list a few just to start out with

File: README.md (hunk 3 above)
Comment (LyonFoster) [reply README.md:37]: is this just indented?

File: README.md (hunk 2 above)
Comment (Tim-Behrer) [reply README.md:23]: I think that we should add 'UAS,Inert,BT' etc or however fit for all the coordinate frames that exist

File: README.md (hunk 3 above)
Comment (aidanwoodcu) [reply README.md:37]: yeah

Comment (LyonFoster) [reply README.md:37]: copy

File: README.md [hunk 4]
```diff
@@ -1,6 +1,10 @@
 # ARDVARC
 Team ARDVARC's github repository
 
+## Team Members (open a pr to put urs below)
+* Tim
+* Lyon
+* Aiden
```
Comment (aidanwoodcu) [inline README.md:7]:
```suggestion
* Aidan
```

File: README.md [hunk 5]
```diff
@@ -11,8 +15,35 @@ Team ARDVARC's github repository
 6. Assign someone to review and complete your pr
 7. Bother them until it goes through
 
-Please Put Your Name Below to indicate that you can do this!
-Tim
+## Vector and Frame Naming Convention
```
Comment (Tim-Behrer) [inline README.md:18]:
```suggestion
## Vector and Frame Naming Convention
#### Coordinate Frames
* `UAS` - denotes the **UAS** relative coordinate frame
* `Inert` - denotes the **Inertial** coordinate frame
* `Cam` - denotes the **Camera** coordinate frame
* `RGV_i` - denotes the (ith) **RGV** coordinate frame
* `BT` - denotes the **Bluetooth** coordinate frame
* `GS` - denoted the **Ground Station** coordinate frame
```

**Review by aidanwoodcu — APPROVED** [review]

**Comment by aidanwoodcu** [comment]
Alright actually there are a few things I don't love about this system now that I'm trying to update my code to match it:
- The `_in` feels redundant for vectors
- I think many vectors are position vectors that are implicitly "from the frame origin to x", but writing that out every time as `vec_origin2x_frame` feels excessive to me.
    - We could make the first part optional, where the frame origin is implied if nothing is specified? So `vec_x_frame` implicitly means "from the frame origin to x"
    - We could add something like `pos_x_frame` for position vectors?
- The `a2b` style doesn't work well for certain types of vectors like velocity vectors or angular velocity vectors where they aren't really from an `a` to a `b`. That's probably fine if we just allow the `a2b` part to be something describing the vector
    - Examples:
        - `vec_uasAngularVelocity_uasBody`
        - `vec_uasHeading_ned`

**Review by Tim-Behrer — APPROVED** [review]

**Comment by LyonFoster** [comment]
I feel like we shouldn't be merging other people's PRs. 

Also to your points Aidan,
* You're right, let's get rid of `in_`
* I think it's worth being explicit about `origin` since positional vectors are always relative vectors
* I'm worried `pos` might  read as pose
* I think that the reason we should be explicit about `origin` is because some vectors (like you point out) aren't relative and those should be the cases when we're just descriptive about what the vector is

**Pull request merged**